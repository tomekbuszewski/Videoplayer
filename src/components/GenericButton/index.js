/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-27
 */

import * as React from 'react';
import inject from 'react-jss';

import styles from './index.styles';

type Props = {
  active: boolean,
  activeState: any,
  classes: Object,
  inactiveState: any,

  onClick: Function,
}

const GenericButton = (props: Props) => {
  const {
    active,
    activeState,
    classes,
    inactiveState,
    onClick,
  } = props;

  return <button className={classes.GenericButton} type="button" onClick={onClick}>{active ? activeState : inactiveState}</button>;
};

export default inject(styles)(GenericButton);
