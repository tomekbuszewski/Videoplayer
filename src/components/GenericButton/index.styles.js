import { STEP_SMALL, STEP_MEDIUM, GET_STEP } from '@config/visuals';
import { COLOR_WHITE } from '@config/colors';

export default {
  GenericButton: {
    padding: GET_STEP(STEP_SMALL),
    background: 'none',
    border: 0,
    color: COLOR_WHITE,

    '& svg': {
      width: GET_STEP(STEP_MEDIUM),
      fill: COLOR_WHITE,
    },
  },
};
