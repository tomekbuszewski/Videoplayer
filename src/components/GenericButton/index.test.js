import * as React from 'react';
import renderer from 'react-test-renderer';

import GenericButton from '.';

describe('GenericButton tests', () => {
  it('should match snapshot while active', () => {
    const component = renderer.create(<GenericButton
      active
      activeState="active"
      inactiveState="inactive"
      onClick={() => false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot while inactive', () => {
    const component = renderer.create(<GenericButton
      active={false}
      activeState="active"
      inactiveState="inactive"
      onClick={() => false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
