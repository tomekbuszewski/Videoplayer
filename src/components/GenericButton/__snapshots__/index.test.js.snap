// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericButton tests should match snapshot while active 1`] = `
<button
  className="GenericButton-GenericButton-0-1-1"
  onClick={[Function]}
  type="button"
>
  active
</button>
`;

exports[`GenericButton tests should match snapshot while inactive 1`] = `
<button
  className="GenericButton-GenericButton-0-1-1"
  onClick={[Function]}
  type="button"
>
  inactive
</button>
`;
